core/src/main/java/org/mule/runtime/core/internal/message/DefaultMessageBuilder.java

```diff
@@ -49,6 +52,11 @@ public final class DefaultMessageBuilder
   private TypedValue payload = of(NULL_TYPED_VALUE);
   private TypedValue attributes = of(NULL_TYPED_VALUE);
 
+  private final Map<String, TypedValue<Serializable>> inboundProperties = new CaseInsensitiveMapWrapper<>();
+  private final Map<String, TypedValue<Serializable>> outboundProperties = new CaseInsensitiveMapWrapper<>();
+  private Map<String, DataHandler> inboundAttachments = new LinkedHashMap<>();
+  private Map<String, DataHandler> outboundAttachments = new LinkedHashMap<>();
```

Comment:
these should not be added back. If they are missing, that should be handled by the serializer